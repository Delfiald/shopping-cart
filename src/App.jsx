import { useEffect, useState } from "react";
import { Outlet } from "react-router-dom";
import Footer from "./components/Footer/Footer";
import { fetchData } from "./services/fakeStoreAPI";
import Header from "./components/Header/Header";
import { getItem } from "./utils/localStorage";
import Feedback from "./components/Feedback/Feedback";

function App() {
 const [cartItem, setCartItem] = useState([]);
 const [wishlistItem, setWishlistItem] = useState([]);
 const [notificationItem, setNotificationItem] = useState([]);
 const [hoverButton, setHoverButton] = useState(null);
 const [searchInput, setSearchInput] = useState("");

 const [error, setError] = useState(null);
 const [loading, setLoading] = useState(null);

 const [categories, setCategories] = useState([]);
 const [products, setProducts] = useState([]);

 const handleProducts = (productsData) => {
  setProducts(productsData);
 };

 const handleCategories = (categoriesData) => {
  setCategories(categoriesData);
 };

 useEffect(() => {
  const fetchDataFromAPI = async () => {
   const productsData = await fetchData(setError, setLoading).getAllProducts();
   const categoriesData = await fetchData(
    setError,
    setLoading
   ).getProductsCategory();
   handleProducts(productsData);
   handleCategories(categoriesData);
  };

  fetchDataFromAPI();
 }, []);

 useEffect(() => {
  const savedCart = getItem("cart");
  const savedWishlist = getItem("wishlist");
  const savedNotification = getItem("notification");

  if (savedCart && Array.isArray(savedCart)) {
   setCartItem(savedCart);
  }

  if (savedWishlist && Array.isArray(savedWishlist)) {
   setWishlistItem(savedWishlist);
  }

  if (savedNotification && Array.isArray(savedNotification)) {
   setNotificationItem(savedNotification);
  }
 }, []);

 return (
  <>
   <Header
    products={products}
    cartItem={cartItem}
    wishlistItem={wishlistItem}
    notificationItem={notificationItem}
    setNotificationItem={setNotificationItem}
    hoverButton={hoverButton}
    setHoverButton={setHoverButton}
    searchInput={searchInput}
    setSearchInput={setSearchInput}
   />
   <Outlet
    context={{
     categories,
     products,
     hoverButton,
     setHoverButton,
     cartItem,
     setCartItem,
     wishlistItem,
     setWishlistItem,
     setNotificationItem,
     searchInput,
    }}
   />
   <Feedback error={error} setError={setError} loading={loading} />
   <Footer />
  </>
 );
}

export default App;
