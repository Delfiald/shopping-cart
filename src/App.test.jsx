import { render } from "@testing-library/react";
import App from "./App";

describe("Something truthy and falsy", () => {
 it("true to be true", () => {
  expect(true).toBe(true);
 });

 it("false to be false", () => {
  expect(false).toBe(false);
 });

 it("render app", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
 });
});
