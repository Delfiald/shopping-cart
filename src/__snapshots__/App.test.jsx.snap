// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`It Render App > render app 1`] = `
<div>
  <header>
    <nav>
      <div
        class="_hero_d33141"
      >
        <h1>
          Shoppers
        </h1>
      </div>
      <div
        class="_search-container_d33141"
      >
        <label
          for="search"
        >
          <svg
            class="lucide lucide-search"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            title="Search Icon"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <path
              d="m21 21-4.3-4.3"
            />
          </svg>
        </label>
        <input
          id="search"
          placeholder="Search Here..."
          type="text"
          value=""
        />
      </div>
      <div
        class="_cart-button-wrapper_d33141"
        data-testid="cart-button-wrapper"
      >
        <button
          class="_cart-button-wrapper_d33141"
          data-testid="cart-button"
        >
          <svg
            class="lucide lucide-shopping-cart"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            title="Cart Button"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="8"
              cy="21"
              r="1"
            />
            <circle
              cx="19"
              cy="21"
              r="1"
            />
            <path
              d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
            />
          </svg>
        </button>
      </div>
      <div
        class="_notification-button-wrapper_d33141"
        data-testid="notification-button-wrapper"
      >
        <button
          class="_notification-button_d33141"
          data-testid="notification-button"
        >
          <svg
            class="lucide lucide-bell"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.268 21a2 2 0 0 0 3.464 0"
            />
            <path
              d="M3.262 15.326A1 1 0 0 0 4 17h16a1 1 0 0 0 .74-1.673C19.41 13.956 18 12.499 18 8A6 6 0 0 0 6 8c0 4.499-1.411 5.956-2.738 7.326"
            />
          </svg>
        </button>
      </div>
    </nav>
  </header>
  <main>
    <div
      class="_hero_716e15"
    >
      <div
        class="_carousel_716e15"
        data-testid="products-carousel-section"
      />
    </div>
    <div
      class="_categories_716e15"
      data-testid="categories-section"
    >
      <h2>
        Categories
      </h2>
    </div>
    <div
      class="_call-to-action_716e15"
    >
      <img
        alt="Call to Action Background"
        src=""
      />
      <button>
        Shop Now
      </button>
    </div>
  </main>
  <footer>
    <div
      class="_top-footer_1a5889"
    >
      <div
        class="content"
        data-testid="about"
      >
        <div
          class="title"
        >
          Shoppers
        </div>
        <div
          class="about"
        >
          About
        </div>
        <div
          class="career"
        >
          Career
        </div>
        <div
          class="blog"
        >
          Blog
        </div>
        <div
          class="sale"
        >
          Sale
        </div>
      </div>
      <div
        class="content"
        data-testid="help"
      >
        <div
          class="title"
        >
          Guide and Help
        </div>
        <div
          class="about"
        >
          Help
        </div>
        <div
          class="career"
        >
          Terms and Condition
        </div>
        <div
          class="blog"
        >
          Privacy
        </div>
        <div
          class="sale"
        >
          Contact Us
        </div>
      </div>
      <div
        class="content"
        data-testid="socials-media"
      >
        <div
          class="title"
        >
          Follow
        </div>
        <a
          href="https://github.com/Delfiald"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="lucide lucide-github"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            title="Github"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
            />
            <path
              d="M9 18c-4.51 2-5-2-7-2"
            />
          </svg>
        </a>
        <a
          href="https://linkedin.com/in/m-aldi-gunawan"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="lucide lucide-linkedin"
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            title="Linkedin"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
            />
            <rect
              height="12"
              width="4"
              x="2"
              y="9"
            />
            <circle
              cx="4"
              cy="4"
              r="2"
            />
          </svg>
        </a>
      </div>
      <div
        class="content"
        data-testid="apps"
      >
        <div
          class="title"
        >
          Download Apps
        </div>
        <div>
          <img
            alt="Google Play"
            src="/Stores/google-play.svg"
          />
          <p>
            GET IT ON
          </p>
          <p>
            Google Play
          </p>
        </div>
        <div>
          <img
            alt="App Store"
            src="/Stores/app-store.svg"
          />
          <p>
            Download on the
          </p>
          <p>
            App Store
          </p>
        </div>
        <div>
          <img
            alt="AppGallery"
            src="/Stores/huawei-app-gallery.svg"
          />
          <p>
            EXPLORE IT ON
          </p>
          <p>
            AppGallery
          </p>
        </div>
      </div>
    </div>
    <div
      class="_bottom-footer_1a5889"
    >
      <p>
        Â© MAG 2025
      </p>
    </div>
  </footer>
</div>
`;
